### PR TITLE
[SMALL FIX] Do not load metadata for non-persisted directory

### PR DESCRIPTION
**Pull request by peisun1115**
Do not throw exception when loading metadata on a non-existing file/dir.

### CODE REVIEW

**Comment by peisun1115** [comment]
@gpang

File: core/server/src/main/java/alluxio/master/file/FileSystemMaster.java [hunk 1]
```diff
@@ -1784,8 +1784,7 @@ private long loadMetadataAndJournal(LockedInodePath inodePath, LoadMetadataOptio
     UnderFileSystem ufs = resolution.getUfs();
     try {
       if (!ufs.exists(ufsUri.toString())) {
-        throw new FileDoesNotExistException(
-            ExceptionMessage.PATH_DOES_NOT_EXIST.getMessage(path.getPath()));
+        return AsyncJournalWriter.INVALID_FLUSH_COUNTER;
```
Comment (gpang) [inline core/server/src/main/java/alluxio/master/file/FileSystemMaster.java]: When trying to load the metadata for a path, if it does not exist in UFS, why should it be a noop? Throwing `FileDoesNotExistException` seems to make sense in that situation?

**Comment by gpang** [comment]
I left a question about this change. Thanks!

File: core/server/src/main/java/alluxio/master/file/FileSystemMaster.java (hunk 1 above)
Comment (peisun1115) [inline core/server/src/main/java/alluxio/master/file/FileSystemMaster.java]: This happens often when people use MUST_CACHE and call list_status on a directory. What I learned from "effective-java" (Item 57) is that exceptions should not be used for normal control flow which can be slow.  I can change this to return a different Exception (say FileDoesNotExistInUFSException) if you prefer. Then I can remove the logging based on this.

Comment (calvinjia) [inline core/server/src/main/java/alluxio/master/file/FileSystemMaster.java]: In that case is a better fix to not try to load metadata for directories which are not persisted or files which are already in Alluxio?

Comment (peisun1115) [inline core/server/src/main/java/alluxio/master/file/FileSystemMaster.java]: done. thanks for the suggestion.

**Comment by peisun1115** [comment]
@gpang @calvinjia  I have updated the pr based your comments. Thanks.

**Comment by gpang** [comment]
LGTM

File: core/server/src/main/java/alluxio/master/file/FileSystemMaster.java [hunk 2]
```diff
@@ -1906,7 +1906,9 @@ private long loadMetadataIfNotExistAndJournal(LockedInodePath inodePath,
     boolean loadDirectChildren = false;
     if (inodeExists) {
       try {
-        loadDirectChildren = inodePath.getInode().isDirectory() && options.isLoadDirectChildren();
+        Inode inode = inodePath.getInode();
+        loadDirectChildren =
+            inode.isDirectory() && inode.isPersisted() && options.isLoadDirectChildren();
```
Comment (apc999) [inline core/server/src/main/java/alluxio/master/file/FileSystemMaster.java]: 1. could you also update the javadoc
2. this function is so important and seems tricky, could you add a unit test for this? (not necessary in this PR)

**Comment by aaudiber** [comment]
@peisun1115 could you update the PR title to reflect what the PR does?

**Comment by calvinjia** [comment]
LGTM

File: core/server/src/main/java/alluxio/master/file/FileSystemMaster.java (hunk 2 above)
Comment (peisun1115) [inline core/server/src/main/java/alluxio/master/file/FileSystemMaster.java]: done

**Comment by peisun1115** [comment]
@apc999  I have added a TOD for the unittest

**Comment by apc999** [comment]
LGTM

**Pull request merged**